pkg/chromium/browser.go

```diff
@@ -176,3 +302,105 @@ func printPDF(requestOptions RenderingOptions, dst chan []byte) chromedp.Action
 		return nil
 	})
 }
+
+func screenshotPNG(requestOptions RenderingOptions, dst chan []byte) chromedp.Action {
+	return chromedp.ActionFunc(func(ctx context.Context) error {
+		output, err := page.CaptureScreenshot().
+			WithFormat(page.CaptureScreenshotFormatPng).
+			WithCaptureBeyondViewport(requestOptions.FullHeight).
+			Do(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to capture screenshot: %w", err)
+		}
+		dst <- output
+		return nil
+	})
+}
+
+func scrollForElements(timeBetweenScrolls time.Duration) chromedp.Action {
+	return chromedp.ActionFunc(func(ctx context.Context) error {
+		var scrolls int
+		err := chromedp.Evaluate(`Math.floor(document.body.scrollHeight / window.innerHeight)`, &scrolls).Do(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to calculate scrolls required: %w", err)
+		}
+
+		time.Sleep(timeBetweenScrolls)
+		for range scrolls {
+			err := chromedp.Evaluate(`window.scrollBy(0, window.innerHeight, { behavior: 'instant' })`, nil).Do(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to scroll: %w", err)
+			}
+			time.Sleep(timeBetweenScrolls)
+		}
+
+		err = chromedp.Evaluate(`window.scrollTo(0, 0, { behavior: 'instant' })`, nil).Do(ctx)
+		if err != nil {
```

Comment:
Why do we need to scroll back to the top? Would it be possible to write a comment explaining?